rfcs/convergence/internationalization.md

```diff
@@ -0,0 +1,217 @@
+
+```js
+var localeValues = {
+  locale: 'es',
+  global: {
+    placeholder: 'Seleccione'
+  },
+  Table: {
+    filterTitle: 'Filtrar menú',
+    emptyText: 'Sin datos',
+    selectAll: 'Seleccionar todo',
+    [...]
+  },
+  Modal: {
+    okText: 'Aceptar',
+    cancelText: 'Cancelar',
+    justOkText: 'Aceptar'
+  }
+  [...etc]
+}
+```
+
+Material UI includes the locale strings in their ThemeProvider, and Ant Design puts it in their ConfigProvider.
+
+Libraries where I found no tools or defined patterns for localization include Lightning Design System, Semantic UI, React Bootstrap, and Carbon Design System.
+
+Two of the most common i18n packages for React independent of component libraries are `react-i18next` (a react wrapper for i18next), and `react-intl`. Both provide a helper function to format a string -- at the most basic, they take a key and return a localized string. The most basic usage looks like this for each library:
+
+react-i18next:
+
+```js
+import { useTranslation } from 'react-i18next';
+
+const Component = () => {
+  const { t } = useTranslation();
+
+  return <div>{t('key')}</div>;
+};
+```
+
+react-intl:
+
+```js
+import { useIntl } from 'react-intl';
+
+const Component = () => {
+  const intl = useIntl();
+
+  return <div>{intl.formatMessage({ id: 'key' })}</div>;
+};
+```
+
+Both also provide extras like React components as an alternative to the function, and helpers for things like dates, string template value replacement, SSR, etc. Those likely aren't relevant to Fluent, since we'll either not use them in the case of the component or SSR, or provide our own solution for dates/string templates.
+
+## Proposal
+
+While we don't have plans to provide actual translations for the strings in any of our components, it'd be nice to create a pattern for defining custom strings that makes it easy for authors to:
+
+- Easily propagate app-level locale strings down to Fluent components without needing to manually map them to multiple separate component props
+- Easily include all necessary component strings when they create a new locale definition/JSON file (i.e. not need to manually remember that TagPicker needs `selectionAriaLabel` defined).
+- Use 3rd party i18n tools with Fluent components without too much extra work
+
+To do this, I think it makes sense to add locale data to Fluent's `ProviderContext`, since it already includes `dir`.
+
+One straightforward option for how to do this would be to provide a locale object with static key/value pairs for each string. We could provide a minimal set of global strings in the top-level provider, similar to how Material and Ant differentiate global vs. component strings. Based on current Fluent v8 and N\* strings, it would look like this:
+
+react-shared-contexts/src/ProviderContext/defaultStrings.ts
+
+```js
+export const defaultLocale {
```

Comment:
I'm not sure how feasible it would be to maintain a global locale string object with our current component architecture.

Each of our components are encapsulated in their own packages, it makes perfect sense to colocate the strings with the components that use them.

By creating a global string object, we need to hoist another context above all other components to share strings, which breaks down the multi-package architecture a bit more. Tree shaking would also need to be considered since we probably would not want components to bundle strings that are not used